src/base/http/server.cpp

```diff
@@ -56,13 +56,32 @@ namespace
     QList<QSslCipher> safeCipherList()
     {
         const QStringList badCiphers {u"idea"_qs, u"rc4"_qs};
+        // Contains Ciphersuites that use RSA for the Key Exchange but they don't mention it in their name
+        const QStringList badRSAShorthandSuites {u"AES256-GCM-SHA384"_qs, u"AES128-GCM-SHA256"_qs,
+                                                                       u"AES256-SHA256"_qs, u"AES128-SHA256"_qs,
+                                                                       u"AES256-SHA"_qs, u"AES128-SHA"_qs};
+        // Contains Ciphersuites that use AES CBC mode but they don't mention it in their name
+        const QStringList badAESShorthandSuites {u"ECDHE-ECDSA-AES256-SHA384"_qs, u"ECDHE-RSA-AES256-SHA384"_qs,
+                                                                     u"DHE-RSA-AES256-SHA256"_qs, u"ECDHE-ECDSA-AES128-SHA256"_qs,
+                                                                     u"ECDHE-RSA-AES128-SHA256"_qs, u"DHE-RSA-AES128-SHA256"_qs,
+                                                                     u"ECDHE-ECDSA-AES256-SHA"_qs, u"ECDHE-RSA-AES256-SHA"_qs,
+                                                                     u"DHE-RSA-AES256-SHA"_qs, u"ECDHE-ECDSA-AES128-SHA"_qs,
+                                                                     u"ECDHE-RSA-AES128-SHA"_qs, u"DHE-RSA-AES128-SHA"_qs};
         const QList<QSslCipher> allCiphers {QSslConfiguration::supportedCiphers()};
         QList<QSslCipher> safeCiphers;
-        std::copy_if(allCiphers.cbegin(), allCiphers.cend(), std::back_inserter(safeCiphers), [&badCiphers](const QSslCipher &cipher)
+        std::copy_if(allCiphers.cbegin(), allCiphers.cend(), std::back_inserter(safeCiphers), [&badCiphers, &badRSAShorthandSuites, &badAESShorthandSuites](const QSslCipher &cipher)
         {
-            return std::none_of(badCiphers.cbegin(), badCiphers.cend(), [&cipher](const QString &badCipher)
+            return std::none_of(badCiphers.cbegin(), badCiphers.cend(), [&cipher, &badRSAShorthandSuites, &badAESShorthandSuites](const QString &badCipher)
             {
-                return cipher.name().contains(badCipher, Qt::CaseInsensitive);
+                const QString name = cipher.name();
+                return name.contains(badCipher, Qt::CaseInsensitive)
+                       || name.contains(u"-cbc-"_qs, Qt::CaseInsensitive) // AES CBC mode is considered vulnerable to BEAST attack
```

Comment:
This and the following tests (that are independent from `badCipher`) should be performed outside from the callback passed in `std::none_of` (before it).

Reply:
You're absolutely correct. I don't know how I missed that.